### PR TITLE
ENH: Fine calibration support for more MEG systems

### PR DESCRIPTION
**Pull request by larsoner**
I recently tried running fine calibration calculation on a non-Neuromag system. It did not work. This PR makes it so it works much better for other systems.

### CODE REVIEW

**Comment by larsoner** [comment]
@drammock is this PR okay with you? It now contains a workaround for https://github.com/scikit-learn/scikit-learn/pull/30268#issuecomment-2479701651 that is causing pip-pre to fail

File: mne/preprocessing/_fine_cal.py [hunk 1]
```diff
@@ -43,6 +44,9 @@ def compute_fine_calibration(
     origin=(0.0, 0.0, 0.0),
     cross_talk=None,
     calibration=None,
+    *,
+    angle_limit=5.0,
+    err_limit=5.0,
```
Comment (drammock) [inline mne/preprocessing/_fine_cal.py:49]: AFAICT `err_limit` isn't validated anywhere, there should at least be some check that it is non-negative right?

File: mne/preprocessing/tests/test_fine_cal.py [hunk 2]
```diff
@@ -126,16 +184,104 @@ def test_compute_fine_cal():
...
+    ],
+)
+def test_fine_cal_systems(system, tmp_path):
+    """Test fine calibration with different systems."""
+    int_order = 8
+    n_ref = 0
+    if system == "kit":
+        raw = read_raw_kit(sqd_path, mrk_path, elp_path, hsp_path)
+        angle_limit = 170
+        err_limit = 500
```
Comment (drammock) [inline mne/preprocessing/tests/test_fine_cal.py:218]: err_limit is 500% for kit, and 6000% for CTF?

**Review by drammock — APPROVED** [review]

File: mne/preprocessing/tests/test_fine_cal.py (hunk 2 above)
Comment (larsoner) [reply mne/preprocessing/tests/test_fine_cal.py:218]: Yeah I think because these are actually not great segments to be processing!

**Pull request merged**